homeassistant/components/hassio/jobs.py

```diff
@@ -158,3 +166,9 @@ def _process_job_change(self, job: Job) -> None:
         for sub in self._subscriptions:
             if sub.matches(job):
                 sub.event_callback(job)
+
+    @callback
+    def unload(self) -> None:
+        """Unregister with dispatcher on config entry unload."""
+        if self._dispatcher_disconnect:
+            self._dispatcher_disconnect()
```

Comment:
Maybe add a test of this where we check that `self._supervisor_client.jobs.info` is only called once after a reload on a Supervisor event.